modules/acquia_telemetry/src/Telemetry.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Drupal\acquia_telemetry;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Component\Utility\Crypt;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Extension\ModuleExtensionList;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Telemetry service.
+ */
+class Telemetry {
```

Comment:
Let's make this class final, and all its protected members private. This is another class we should draw sharp boundaries around. (In general, that's the policy I'm taking towards Lightning development; classes should be final and private unless they have a specific reason not to be -- this allows us to have _intentional_ APIs, rather than accidental ones.)

Reply:
Trouble with making this final is that you can't then create a prophecy for the call when testing. Any suggestions?